addOns/llm/src/main/java/org/zaproxy/addon/llm/ui/ImportDialog.java

```diff
@@ -0,0 +1,311 @@
+        row++;
+        add(getCancelButton(), LayoutHelper.getGBC(0, row, 1, 0.5, new Insets(4, 8, 8, 4)));
+        add(getImportButton(), LayoutHelper.getGBC(1, row, 1, 0.5, new Insets(4, 4, 8, 8)));
+        row++;
+        add(getProgressBar(), LayoutHelper.getGBC(0, row, 2, 1.0, new Insets(0, 8, 8, 8)));
+        pack();
+        setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+    }
+
+    private boolean importOpenapi()
+            throws IOException, URISyntaxException, ApiException, DatabaseException {
+
+        String openapiLocation = getOpenapiField().getText();
+        llmOptionsParam = extLlm.getOptionsParam();
+        Integer endpointCount = 0;
+
+        if (StringUtils.isEmpty(llmOptionsParam.getApiKey())) {
+            showWarningDialog(Constant.messages.getString("llm.options.apikey.error.undefinded"));
+            throw new RuntimeException(
+                    Constant.messages.getString("llm.options.apikey.error.undefinded"));
+        }
+
+        if (StringUtils.isEmpty(llmOptionsParam.getEndpoint())) {
+            showWarningDialog(Constant.messages.getString("llm.options.endpoint.error.undefinded"));
+            throw new RuntimeException(
+                    Constant.messages.getString("llm.options.endpoint.error.undefinded"));
+        }
+
+        LlmCommunicationService llmCommunicationService =
+                new LlmCommunicationService(
+                        llmOptionsParam.getModelName(),
+                        llmOptionsParam.getApiKey(),
+                        llmOptionsParam.getEndpoint());
+
+        if (StringUtils.isEmpty(openapiLocation)) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        "llm.importDialog.error.missingOpenapi"));
+                        getOpenapiField().requestFocusInWindow();
+                    });
+            return false;
+        }
+
+        try {
+            new URL(openapiLocation).toURI();
+            new URI(openapiLocation, true);
+            endpointCount = llmCommunicationService.importOpenapiFromUrl(openapiLocation);
+
+            return true;
+        } catch (URIException | MalformedURLException | URISyntaxException e) {
+            // Not a valid URI, try to import as a file
+            endpointCount = llmCommunicationService.importOpenapiFromFile(openapiLocation);
+        }
+
+        var file = new File(openapiLocation);
+        if (!file.canRead()) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningFileNotFound(openapiLocation);
+                        getOpenapiField().requestFocusInWindow();
+                    });
+            return false;
+        }
+
+        endpointCount = llmCommunicationService.importOpenapiFromFile(openapiLocation);
```

Comment:
This should probably be wrapped in a condition, in order to not override the operations above.